source/common/router/route_config_update_receiver_impl.cc

```diff
@@ -32,13 +32,25 @@ bool RouteConfigUpdateReceiverImpl::onVhdsUpdate(
     const Protobuf::RepeatedPtrField<envoy::api::v2::Resource>& added_resources,
     const Protobuf::RepeatedPtrField<std::string>& removed_resources,
     const std::string& version_info) {
+  // TODO: check if any changes have been made
+  collectAliasesInUpdate(added_resources);
   removeVhosts(virtual_hosts_, removed_resources);
   updateVhosts(virtual_hosts_, added_resources);
   rebuildRouteConfig(virtual_hosts_, route_config_proto_);
 
   return onRdsUpdate(route_config_proto_, version_info);
 }
 
+void RouteConfigUpdateReceiverImpl::collectAliasesInUpdate(
+    const Protobuf::RepeatedPtrField<envoy::api::v2::Resource>& added_resources) {
+  aliases_in_last_update_.clear();
+  for (const auto& resource : added_resources) {
+    for (const std::string& alias : resource.aliases()) {
```

Comment:
I think this inner loop can become something like `std::copy(resource.aliases().begin(), resource.aliases().end(), std::back_inserter(aliases_in_last_update_));`. Actually I think what you have is more readable, so if you want to keep it as-is I won't complain, but I think the general preference is to use these things when possible.

Reply:
done.